### PR TITLE
chore: update pnpm version to 9.6.0

### PR DESCRIPTION
**Pull request by sukvvon**
**What**:

* update pnpm version to 9.6.0

**Why**:

* pnpm version is outdated

**How**:

* update pnpm version to up to date

**Checklist**:

- [x] Documentation
- [x] Tests
- [x] Ready to be merged

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "description": "Simple reusable React error boundary component",
   "author": "Brian Vaughn <brian.david.vaughn@gmail.com>",
   "license": "MIT",
-  "packageManager": "pnpm@8.6.10",
+  "packageManager": "pnpm@9.6.0",
```
Comment (sukvvon) [inline package.json:7]: declare packageManager version like [pnpm's package.json](https://github.com/pnpm/pnpm/blob/main/package.json#L66)

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ jobs:
       - uses: actions/checkout@v4
       - uses: pnpm/action-setup@v4
         with:
-          version: 8.6.10
+          version: 9.6.0
```
Comment (sukvvon) [inline .github/workflows/ci.yml:17]: match pnpm/action-setup@v4's version with package.json's packageManager version

**Comment by bvaughn** [comment]
I think the `"engines"` block is a nicer way to do this, since it will error if you try running with a different version. It also looks like our package locks are pretty different ([see here](https://github.com/sukvvon/react-error-boundary/compare/chore/update-pnpm-version...bvaughn:react-error-boundary:pnpm9)).

I think I'll just do this upgrade via #198 if that's okay.

**Comment by sukvvon** [comment]
@bvaughn i think we can solve #201 with this pr

**Pull request closed without merging**